esx_service/vmdk_ops.py

```diff
@@ -844,18 +902,19 @@ def err(string):
     return {u'Error': string}
 
 
-def disk_detach(vmdk_path, vm):
+def disk_detach(vmdk_path, vm, device=None):
```

Comment:
The device doesn't need to be passed in here. It results in 2 extra checks.  Could just call `return disk_detach(..)` directly from `handle_stale_attach` and remove all the checking and message building code. Also, if you do it this way you get the warning log as well, which isn't in the `handle_stale_attach` code path.